### PR TITLE
Add toLastDayOfMonth function

### PR DESCRIPTION
**Pull request by holadepo**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `toEndOfMonth` function which rounds up a date or date with time to the last day of the month. #33501

### CODE REVIEW

**Comment by novikd** [comment]
@Mergifyio update

File: src/Common/tests/gtest_DateLUTImpl.cpp [hunk 1]
```diff
@@ -264,6 +268,8 @@ TEST(DateLUTTest, TimeValuesAtRightBoderOfRangeOfOldLUT)
 
     EXPECT_EQ(lut.timeToString(time), "2106-01-31 01:17:53" /*std::string*/);
     EXPECT_EQ(lut.dateToString(time), "2106-01-31" /*std::string*/);
+    EXPECT_EQ(lut.toLastDayOfMonth(time), 4294339200 /*time_t*/); // 2016-01-01
```
Comment (novikd) [inline src/Common/tests/gtest_DateLUTImpl.cpp:271]: Strange comment

**Review by novikd — APPROVED** [review]
I'd recommend to add a test with some date in February.

File: src/Common/tests/gtest_DateLUTImpl.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Common/tests/gtest_DateLUTImpl.cpp]: It is a typo, copy-pasted from other place.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Functional tests are missing.
We respect unit tests, but functional (SQL) tests are primary.

**Comment by alexey-milovidov** [comment]
@holadepo Do you want to add functional tests?

**Comment by holadepo** [comment]
> @holadepo Do you want to add functional tests?

Yes I will do that

**Comment by alexey-milovidov** [comment]
Should we wait a couple of more days?

**Comment by holadepo** [comment]
> Should we wait a couple of more days?

Yes please. Give me till Monday, 4th April.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**